overlord/snapstate/snapstate.go

```diff
@@ -531,6 +531,20 @@ func doInstall(st *state.State, snapst *SnapState, snapsup *SnapSetup, flags int
 		prev = installHook
 	}
 
+	if snapsup.QuotaGroupName != "" {
+		// This could result in doing 'setup-profiles' twice, but
+		// unfortunately we can't execute this code earlier as the snap
+		// needs to appear as installed first.
+		// XXX: Could we maybe move 'setup-profiles' to after this somehow? Or
+		// maybe just accept this could incur 'setup-profiles' twice?
```

Comment:
no, we can't move the whole setup-profiles, as it needs to happen before link-snap. setup-profiles uses Ensure* helpers quite a bit so disk writing is not a big issue, but is still a bit annoying. We need to look into this later